src/processor/operator/ddl/add_rel_property.cpp

```diff
@@ -0,0 +1,121 @@
+        fillInMemColumWithDefaultVal(
+            inMemColumn.get(), getDefaultVal(), nodeStatistics->getMaxNodeOffset());
+    }
+    inMemColumn->saveToFile();
+}
+
+void AddRelProperty::createFileForListsProperty(table_id_t boundTableID, RelDirection direction,
+    property_id_t propertyID, RelStatistics* relStatistics) {
+    auto inMemList = InMemListsFactory::getInMemPropertyLists(
+        StorageUtils::getRelPropertyListsFName(storageManager.getDirectory(), tableID, boundTableID,
+            direction, propertyID, DBFileType::ORIGINAL),
+        dataType, relStatistics->getNumTuples());
+    auto boundNodeStatistics = storageManager.getNodesStore()
+                                   .getNodesStatisticsAndDeletedIDs()
+                                   .getNodeStatisticsAndDeletedIDs(boundTableID);
+    // Note: we need the listMetadata to get the num of elements in a large list, and headers to
+    // get the num of elements in a small list as well as determine whether a list is large or
+    // small. All property lists share the same listHeader which is stored in the adjList.
+    auto adjLists = storageManager.getRelsStore().getAdjLists(direction, boundTableID, tableID);
+    inMemList->initListsMetadataAndAllocatePages(boundNodeStatistics->getNumTuples(),
+        adjLists->getHeaders().get(), &adjLists->getListsMetadata());
+    if (!isDefaultValueNull()) {
+        fillInMemListWithDefaultVal(inMemList.get(), getDefaultVal(),
+            boundNodeStatistics->getMaxNodeOffset(),
+            storageManager.getRelsStore().getAdjLists(direction, boundTableID, tableID));
+    }
+    inMemList->saveToFile();
+}
+
+void AddRelProperty::fillInMemListsWithStrValFunc(InMemLists* inMemLists, uint8_t* defaultVal,
+    PageByteCursor& pageByteCursor, node_offset_t nodeOffset, list_header_t header,
+    uint64_t posInList) {
+    auto strVal = *(ku_string_t*)defaultVal;
+    inMemLists->getInMemOverflowFile()->copyStringOverflow(
+        pageByteCursor, reinterpret_cast<uint8_t*>(strVal.overflowPtr), &strVal);
+    inMemLists->setElement(header, nodeOffset, posInList, reinterpret_cast<uint8_t*>(&strVal));
+}
+
+void AddRelProperty::fillInMemListsWithListValFunc(InMemLists* inMemLists, uint8_t* defaultVal,
+    PageByteCursor& pageByteCursor, node_offset_t nodeOffset, list_header_t header,
+    uint64_t posInList) {
+    auto listVal = *reinterpret_cast<ku_list_t*>(defaultVal);
+    inMemLists->getInMemOverflowFile()->copyListOverflow(
+        pageByteCursor, &listVal, dataType.childType.get());
+    inMemLists->setElement(header, nodeOffset, posInList, reinterpret_cast<uint8_t*>(&listVal));
+}
+
+fill_in_mem_lists_function_t AddRelProperty::getFillInMemListsFunc() {
+    switch (dataType.typeID) {
+    case INT64:
+    case DOUBLE:
+    case BOOL:
+    case DATE:
+    case TIMESTAMP:
+    case INTERVAL: {
+        return &AddRelProperty::fillInMemListsWithNonOverflowValFunc;
```

Comment:
I guess you don't need & if the function becomes static.

Reply:
removed